sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -243,13 +243,61 @@ namespace Azure { namespace Core { namespace Test {
           "Test Log from: [ " + m_testContext.GetTestPlaybackRecordingName() + " ] - " + message);
     }
 
-    // Util for tests getting env vars
-    std::string GetEnv(const std::string& name)
+    /**
+     * @brief Utility function used by tests to retrieve env vars
+     *
+     * @param name Environment variable name to retrieve.
+     *
+     * @return The value of the environment variable retrieved.
+     *
+     * @note If AZURE_TENANT_ID, AZURE_CLIENT_ID, or AZURE_CLIENT_SECRET are not available in the
+     * environment, the AZURE_SERVICE_DIRECTORY environment variable is used to set those values
+     * with the values emitted by the New-TestResources.ps1 script.
+     *
+     * @note The Azure CI pipeline upper cases all environment variables defined in the pipeline.
+     * Since some operating systems have case sensitive environment variables, on debug builds, this
+     * function ensures that the environment variable being retrieved is all upper case.
+     *
+     */
+    std::string GetEnv(std::string const& name)
     {
-      const auto ret = Azure::Core::_internal::Environment::GetVariable(name.c_str());
-
+#if !defined(NDEBUG)
+      // The azure CI pipeline uppercases all EnvVar values from ci.yml files.
+      // That means that any mixed case strings will not be found when run from the CI
+      // pipeline. Check to make sure that the developer only passed in an upper case environment
+      // variable.
+      {
+        if (name != Azure::Core::_internal::StringExtensions::ToUpper(name))
+        {
+          throw std::runtime_error("All Azure SDK environment variables must be all upper case.");
+        }
+      }
+#endif
+      auto ret = Azure::Core::_internal::Environment::GetVariable(name.c_str());
       if (ret.empty())
       {
+        if (!m_testContext.IsPlaybackMode() && name.find("AZURE_") == 0)
```

Comment:
nit.
Move "AZURE_" to a const variable and re-use it below on line 294 for calling substr()